addons/account/models/account_payment_term.py

```diff
@@ -21,6 +22,22 @@ def _default_line_ids(self):
     company_id = fields.Many2one('res.company', string='Company')
     sequence = fields.Integer(required=True, default=10)
 
+    # -------Early payment discount fields-------
+    has_early_payment = fields.Boolean(string=_("Apply Early Payment Discount"), default=False)
```

Comment:
you don't need explicit translation in fields
```suggestion
    has_early_payment = fields.Boolean(string="Apply Early Payment Discount")
```
(a boolean field is already false by default)